node_modules/oae-authentication/lib/api.js

```diff
@@ -270,25 +272,63 @@ var getOrCreateUser = module.exports.getOrCreateUser = function(ctx, authProvide
  * @api private
  */
 var _getOrCreateUser = function(ctx, loginId, displayName, opts, callback) {
+    opts = opts || {};
+
     _getUserIdFromLoginId(loginId, function(err, userId) {
         if (err && err.code !== 404) {
             return callback(err);
         } else if (!userId) {
-            // No user mapped to this login id so we will create a user for it
-            log(ctx).trace({
-                'authProvider': loginId.provider,
-                'externalId': loginId.externalId,
-                'displayName': displayName,
-                'opts': opts
-            }, 'Auto-creating a user on login');
-
-            createUser(ctx, loginId, displayName, opts, function(err, user) {
-                if (err) {
-                    return (err);
-                }
 
-                return callback(null, user, _flattenLoginId(loginId), true);
-            });
+            var _createAndReturn = function() {
+                // No user mapped to this login id so we will create a user for it
+                log(ctx).trace({
+                    'authProvider': loginId.provider,
+                    'externalId': loginId.externalId,
+                    'displayName': displayName,
+                    'opts': opts
+                }, 'Auto-creating a user on login');
+
+                createUser(ctx, loginId, displayName, opts, function(err, user) {
+                    if (err) {
+                        return callback(err);
+                    }
+
+                    return callback(null, user, _flattenLoginId(loginId), true);
+                });
+            };
+
+            // If we got an email address, we try to look up the user by their email address
+            if (opts.email) {
+                PrincipalsDAO.getUserViaEmail(opts.email, function(err, user) {
+                    if (err && err.code !== 404) {
+                        return callback(err);
+                    } else if (err && err.code === 404) {
+                        // The email was not in the database, create the user account
+                        return _createAndReturn();
+                    }
+
+                    // At this point, we found a user with the given email address. If this
+                    // user is trying to create an account from a source we don't trust, we
+                    // bail out of the user creation process
+                    if (opts.emailVerified !== true) {
```

Comment:
We can't trust all authentication strategies to always do the right thing. IIRC, when I changed my email address in Twitter it would return that email to OAE even though I hadn't verified it yet. If you know someone's email address, that would be a potential attack vector to associate a twitter login id to that user account.